app/StateToken.php

```diff
@@ -24,6 +27,53 @@ class StateToken extends Model
     protected $hidden = [
     ];
 
+    /**
+     * Makes and checks for an unused token
+     *
+     * @return string
+    */
+    public static function generateUnusedToken()
+    {
+        do {
+            // TODO: Deal with possibility uuid4() may throw an exception of it's own?
+            try {
+                $candidate = Uuid::uuid4()->toString();
+            } catch (\Exception $e) {
+                // Uuid4() throws exceptions, apparently! Log that and die, I guess?
```

Comment:
What. uuid4() throws? What's the error?

Reply:
"\Ramsey\Uuid\Exception\UnsatisfiedDependencyException if `Moontoast\Math\BigNumber` is not present" in the source.

Reply:
I think it's an optional, but reccomeded _large_ int handler.

Reply:
Oh, how frequently does that happen? You would have thought the lib handled that dep missing better if it was optional